api/APIClient.ts

```diff
@@ -0,0 +1,15 @@
+import { middlewareRunner } from "../lib/Middleware";
+import { NonEmptyArray } from "../lib/Types/HelperTypes";
+import { TiFAPIClient, TiFAPISchema } from "./TiFAPISchema";
+import { APIMiddleware, APISchema } from "./TransportTypes";
+
+export const APIClientCreator = <T>(endpointSchema: APISchema, ...apiMiddleware: NonEmptyArray<APIMiddleware<T>>) => 
+  Object.entries(endpointSchema).reduce((apiClient: any, [endpointName, endpointSchema]) =>
```

Comment:
Nit: We could use the `EndpointSchemasToFunctions` type here to avoid using any. This way, it will also infer the type of the client.

Reply:
Wasn't able to get rid of all the 'any's but it did help remove some of them